AsyncDisplayKit/ASTextNode.mm

```diff
@@ -470,6 +485,8 @@ - (id)_linkAttributeValueAtPoint:(CGPoint)point
    inAdditionalTruncationMessage:(out BOOL *)inAdditionalTruncationMessageOut
                  forHighlighting:(BOOL)highlighting
 {
+  ASDisplayNodeAssertMainThread();
```

Comment:
Worth noting that just because something is on the main thread doesn't mean you don't have to lock on ivar access. A background thread could hypothetically be altering it while we're using the ivars in this method and we'll corrupt the heap.